pkg/probe/grpc/grpc.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/klog/v2"
+	"time"
+
+	"google.golang.org/grpc"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/component-base/version"
+	"k8s.io/kubernetes/pkg/probe"
+)
+
+// Prober is an interface that defines the Probe function for doing GRPC readiness/liveness checks.
+type Prober interface {
+	Probe(host, service string, port int, timeout time.Duration, opts ...grpc.DialOption) (probe.Result, string, error)
+}
+
+type grpcProber struct {
+}
+
+// New Prober for execute grpc probe
+func New() Prober {
+	return grpcProber{}
+}
+
+// Probe executes a grpc call to check the liveness/readiness of container.
+// Returns the Result status, command output, and errors if any.
+func (p grpcProber) Probe(host, service string, port int, timeout time.Duration, opts ...grpc.DialOption) (probe.Result, string, error) {
+	v := version.Get()
+
+	md := metadata.New(map[string]string{
+		"User-Agent": fmt.Sprintf("kube-probe/%s.%s", v.Major, v.Minor),
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, fmt.Sprintf("%s:%d", host, port), opts...)
+
+	if err != nil {
+		klog.ErrorS(err, "GRPC probe failed DialContext")
+		return probe.Failure, "", err
+	}
+
+	defer func() {
+		_ = conn.Close()
+	}()
+
+	client := grpchealth.NewHealthClient(conn)
+
+	resp, err := client.Check(metadata.NewOutgoingContext(ctx, md), &grpchealth.HealthCheckRequest{
+		Service: service,
+	})
+
+	if err != nil {
+		klog.ErrorS(err, "GRPC probe failed make client")
```

Comment:
```suggestion
		klog.ErrorS(err, "GRPC probe failed to make a client")
```

Is it actually what happened? Client was already made, check failed, correct?

Maybe get inspiration from ideas of error messages from here: https://github.com/grpc-ecosystem/grpc-health-probe/blob/e2dfb8e0e46eb7c762abb54792c1f5574b25c635/main.go#L294-L304